forms-flow-nav/src/Navbar.jsx

```diff
@@ -170,7 +170,26 @@ const NavBar = React.memo(({ props }) => {
 
   useEffect(() => {
     fetchSelectLanguages((data) => {
-      setSelectLanguages(data);
+      const tenantdata = JSON.parse(StorageService.get("TENANT_DATA"));
+      const getSupportedLanguages = (data, userLanguageList) => {
+        let userLanguagesArray = [];
+        if (typeof userLanguageList === 'object') {
+           userLanguagesArray = Object.values(userLanguageList);
+        } else if (typeof userLanguageList === 'string') {
+           userLanguagesArray = userLanguageList.split(',');
+        }
+        return data.filter(item => userLanguagesArray.includes(item.name));
+      };
+      if (MULTITENANCY_ENABLED && tenantdata?.details.langList) {
```

Comment:
Please optimise the if and else conditions so that repeated code can be removed

Reply:
okay.